base/irrationals.jl

```diff
@@ -13,7 +13,22 @@ promote_rule{s,T<:Number}(::Type{Irrational{s}}, ::Type{T}) = promote_type(Float
 convert(::Type{AbstractFloat}, x::Irrational) = Float64(x)
 convert(::Type{Float16}, x::Irrational) = Float16(Float32(x))
 convert{T<:Real}(::Type{Complex{T}}, x::Irrational) = convert(Complex{T}, convert(T,x))
-convert{T<:Integer}(::Type{Rational{T}}, x::Irrational) = convert(Rational{T}, Float64(x))
+
+@generated function convert{T<:Integer}(::Type{Rational{T}}, x::Irrational)
+    o = precision(BigFloat)
+    p = 256
+    while true
+        setprecision(BigFloat, p)
+        bx = BigFloat(x())
```

Comment:
right, no, this is a generated function, this is how they're supposed to work which I just keep forgetting